### DISCUSSION TITLE
Problems with slideshow

### DISCUSSION THREAD

**Discussion by Oesterd** [AutoDarkMode/Windows-Auto-Night-Mode#761] [Q&A] [answered]
I'm trying to use Auto Dark Mode along with slideshow setting from Windows 10, however, every time I start Windows, the theme changes automatically to some themes named ADMUnmanaged(Light/Dark), which are created automatically even if I delete them.

EDIT: It's not only after startup, but after the scheduled mode change from light to dark or vice-versa, ADMUnmanaged Light/Dark themes gets automatically selected as well, so I have to manually change it back also.

![image](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/126357573/f9a492bd-7aa8-448a-95b0-c74de5be402b)
![image](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/126357573/43e8c791-3f96-4d53-9d5f-59010e66dab0)

The problem with these themes is that the slideshow doesn't work with them, it just displays the first image in the folder as if it were in the "picture" setting, I never heard about these ADM themes, it seems it's something specific to the app itself since the only result I get when searching about it on Google is this page, although I couldn't find any mention to this here:

![image](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/126357573/57ba2303-574c-48f5-acee-6d1b912dcb03)
 
I also tried to find something about themes changing automatically on Google itself, but nothing really useful, that's why I thought it'd be worth asking here.

I'm using this version of the app:

- Commit: `212dbb4`
- Service/App: `10.3.0.90`
- Updater: `2.5.2`
- Shell: `1.3.1.0`
- .Net: `6.0.10`
- Windows: `19045.3324`

It's a Windows 10 Pro x64, if it's ever useful.

Thanks in advance.

EDIT 2: Seems the ADM themes don't stop the slideshow, I still don't know what they are or why they are created, but I was wrong about the problem. Still, there's still some weird behavior about the slideshow intervals changing whenever I change the theme or the tab in the settings windows, but it's not like I'll go as far to ask for a fix just for this.

**Comment by Spiritreader** [comment]
The ADM unmanaged themes are (supposed to be) carbon copies of the custom themes you defined. 
In theme mode, they are applied instead of the actual themes you selected, because specific tweaks to theme files are necessary in order to make the switch work at all.
They're called unmanaged because Auto Dark Mode is not actively controlling/modifying/managing individual settings, but rather retrieves a user-defined / user-managed theme.

The configured slideshow interval should not be touched, unless there is a variable in the theme file that I didn't account for when I implemented theme file parsing.

You can certainly ask for a fix. If it's related to ADM, I'll try to investigate if I can get it working again.
In order to do that, I would need the .theme file you use with a set interval, and a description for how long the interval is supposed to be in Windows.

Please note though: When you have your light mode slideshow running and the you switch your theme to the dark mode slideshow, the interval timer may reset.

Also, you're not on the latest version of ADM. Switch to the beta branch in settings, update and try again first.

**Reply by Oesterd** [reply]
https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/126357573/10371ca7-6111-4f70-be46-d2f5831e9e1d

Here is the behavior, as you said, when the theme changes the interval resets and because the theme changes whenever I go to any other tab and come back to the Themes tab, the interval changes as well and any changes I make in registry are lost.

The original interval was 5 minutes (300 000 ms), under the 1 minute option in the dropdown menu, for some reason that option changes to 10 minutes when I change the interval to 30s (30 000 ms) in the registry?

Here are the theme files:

[Themes.zip](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/files/12507689/Themes.zip)

This video was made using the previous version of ADM but this behavior persists in the latest version as well.

**Reply by Spiritreader** [reply]
Registry changes most likely won't persist.
ADM will only be able to pick up changes made via the windows UI

**Reply by Spiritreader** [reply, marked as answer]
I've managed to ascertain that Auto Dark Mode parses and sets the interval correctly within the theme file.

However, Windows themes only seem to apply slideshow timing values that are configurable in the UI:
![image](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/18438899/3d5f9d14-f5dd-4a39-b4b4-51397397f52a)

Values out side of that range are not supported. Instead, Windows then selects 1 minute as default option.
Your theme has 5 minutes set, which is not among the supported values. Thus it won't work.

It might be possible for you to write a script that manually changes the registry entry after a theme has been applied and hook that with the ADM scripting engine if you want a custom interval.
